### ISSUE TITLE
Generated pkg-config has wrong order (crypto before ssl)

### ISSUE THREAD

**Issue by ewoutp** [edenhill/librdkafka#1576]
Description
===========

The generated pkg-config file (/usr/lib/pkgconfig/rdkafka.pc) has an invalid ordering.
It has `-lcrypto` before `-lssl`. This cases undefined references when trying to build the go client with `-tags static_all`.

How to reproduce
================

Create a docker image using this build file.

```
FROM golang:1.9.2

RUN \
    apt-get update && \
    apt-get install -y liblz4-1 liblz4-dev libssl-dev libcrypto++-dev && \
    mkdir -p /usr/src && \
    cd /usr/src && \
    git clone -b v0.11.1 https://github.com/edenhill/librdkafka.git && \
    cd librdkafka && \
    ./configure --prefix /usr && \
    make && \
    make install

RUN sed -i -e "s|^Libs.private.*$|Libs.private: -lpthread -lssl -lcrypto -llz4 -lrt -ldl|" "/usr/lib/pkgconfig/rdkafka.pc"
```

Note: The last RUN command fixes the problem locally, without it, the ordering is incorrect.
Remove this line to reproduce the problem.

Now try to build any go program with `-tags static_all` as desceibed in https://github.com/confluentinc/confluent-kafka-go.

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:

 - [*] librdkafka version (release number or git tag): v0.11.1
 - [*] Apache Kafka version: irrelevant
 - [*] librdkafka client configuration: irrelevant
 - [*] Operating system: debian 9 (docker image derived from golang:1.9.2)
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
The order corresponds to the order of checks in configure.librdkafka, where it checks for libcrypto before libssl since the latter depends on the former, but the dependencies are added in the check order.
They should probably be added in reverse order, which should be fixed in mklove/modules/configure.base

**Comment by edenhill** [comment]
fixed on master in a0045686ad408365991a674383799bf48e35e356
